lib/restrict_admin_by_ip_middleware.rb

```diff
@@ -0,0 +1,32 @@
+class RestrictAdminByIpMiddleware
+  def initialize(app, allowed_ips)
+    @app = app
+    @allowed_ips = allowed_ips.map { |ip| IPAddr.new(ip) }
+  end
+
+  def call(env)
+    status, headers, body = @app.call(env)
+
+    if restricted?(env)
+      [403, headers, "Forbidden"]
+    else
+      [status, headers, body]
+    end
+  end
+
+  private
+
+  def restricted?(env)
+    req = Rack::Request.new(env)
+
+    restricted_route?(req) && !allowed_ip?(req)
+  end
+
+  def restricted_route?(req)
+    req.path =~ %r{^/admin}
```

Comment:
Ruby gives us [`String#start_with?`](https://apidock.com/ruby/String/start_with%3F), which might be a more expressive way to perform this check.

Reply:
Good shout.